compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -2547,6 +2544,18 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             "ban_nonexisting_field: field={:?}, base={:?}, expr={:?}, base_ty={:?}",
             ident, base, expr, base_ty
         );
+
+        for (ty, _) in self.autoderef(expr.span, base_ty) {
```

Comment:
What is this trying to achieve?

Reply:
trying to avoid no fields diagnostics for this kind of case:
https://github.com/rust-lang/rust/pull/127575/files#diff-6ae6bc33074732b1c0e778c75e56ea6be0c78894cea7c280423f9c1b84a3e82bR8-R18